examples/custom_notification.rs

```diff
@@ -0,0 +1,194 @@
+            message: message.into(),
+        }
+    }
+}
+
+#[derive(Debug)]
+struct CustomNotification {}
+
+impl Notification for CustomNotification {
+    type Params = CustomNotificationParams;
+    const METHOD: &'static str = "custom/notification";
+}
+
+#[derive(Debug, Default)]
+struct Backend;
+
+impl LanguageServer for Backend {
+    type ShutdownFuture = BoxFuture<()>;
+    type SymbolFuture = BoxFuture<Option<Vec<SymbolInformation>>>;
+    type ExecuteFuture = BoxFuture<Option<Value>>;
+    type CompletionFuture = BoxFuture<Option<CompletionResponse>>;
+    type HoverFuture = BoxFuture<Option<Hover>>;
+    type SignatureHelpFuture = BoxFuture<Option<SignatureHelp>>;
+    type DeclarationFuture = BoxFuture<Option<GotoDefinitionResponse>>;
+    type DefinitionFuture = BoxFuture<Option<GotoDefinitionResponse>>;
+    type TypeDefinitionFuture = BoxFuture<Option<GotoDefinitionResponse>>;
+    type ImplementationFuture = BoxFuture<Option<GotoImplementationResponse>>;
+    type HighlightFuture = BoxFuture<Option<Vec<DocumentHighlight>>>;
+
+    fn initialize(&self, _: &Printer, _: InitializeParams) -> Result<InitializeResult> {
+        Ok(InitializeResult {
+            server_info: None,
+            capabilities: ServerCapabilities {
+                text_document_sync: Some(TextDocumentSyncCapability::Kind(
+                    TextDocumentSyncKind::Incremental,
+                )),
+                hover_provider: Some(true),
+                completion_provider: Some(CompletionOptions {
+                    resolve_provider: Some(false),
+                    trigger_characters: Some(vec![".".to_string()]),
+                    work_done_progress_options: Default::default(),
+                }),
+                signature_help_provider: Some(SignatureHelpOptions {
+                    trigger_characters: None,
+                    retrigger_characters: None,
+                    work_done_progress_options: Default::default(),
+                }),
+                document_highlight_provider: Some(true),
+                workspace_symbol_provider: Some(true),
+                execute_command_provider: Some(ExecuteCommandOptions {
+                    commands: vec![
+                        "dummy.do_something".to_string(),
+                        "custom.notification".to_string(),
+                    ],
+                    work_done_progress_options: Default::default(),
+                }),
+                workspace: Some(WorkspaceCapability {
+                    workspace_folders: Some(WorkspaceFolderCapability {
+                        supported: Some(true),
+                        change_notifications: Some(
+                            WorkspaceFolderCapabilityChangeNotifications::Bool(true),
+                        ),
+                    }),
+                }),
+                ..ServerCapabilities::default()
+            },
+        })
+    }
+
+    fn initialized(&self, printer: &Printer, _: InitializedParams) {
```

Comment:
Maybe to avoid unnecessary noise in this example code you could avoid implementing the `initialized()` and `did_*()` notifications? These methods all provide default implementations that do nothing and can be left unspecified in this example.